private/buf/bufformat/bufformat.go

```diff
@@ -66,3 +68,9 @@ func Format(ctx context.Context, module bufmodule.Module) (_ storage.ReadBucket,
 	}
 	return readWriteBucket, nil
 }
+
+// Formats the given file node, writing to dest.
+func FormatFileNode(dest io.Writer, fileNode *ast.FileNode) error {
```

Comment:
In your usage of `FormatFileNode`, are you effectively duplicating this logic elsewhere?

```
		fileNode, err := parser.Parse(moduleFile.ExternalPath(), moduleFile, reporter.NewHandler(nil))
			if err != nil {
				return err
			}
```

If so, it would seem to make sense to make this signature `FormatFile(output io.Writer, file io.Reader, filePath string) error` and put the parsing within the function.

Regardless, we'll want to update `Format` to use this function directly.

We also likely want to rename `Format` to `FormatModule`.

Reply:
My usage uses protocompile library directly to re-parse the file on every ~keystroke without going through module related code (to provide live parse feedback), so I always have the fileNode, but not necessarily a full module.

Reply:
Are you using the `*ast.FileNode` outside of invoking this function? As in is this the only place you use the `*ast.FileNode`, or do you have other uses for it outside of format?

Reply:
I have lots of other use cases for *ast.FileNode, its the basis for for most of the LSP